configure.ac

```diff
@@ -2413,7 +2413,7 @@ fi
         exit 1
     fi
 
-    if test "x$enable_debug" = "xyes"; then
+    if test "x$enable_rust_debug" = "xyes"; then
```

Comment:
maybe this should check for either of those 2 vars, because we want the debug build for rust also if rust-debug isn't enabled. In the old scheme, --enable-debug will make sure symbols are preserved in the rust binary iirc. Do I remember correctly @jasonish ?

Reply:
We keep debug symbols in the release binary as well due to having this in the Cargo.toml:
```
[profile.release]
debug = true
```
So this just enabled/disables optimized builds.

Do we need a specific flag for Rust debug builds anymore?

Reply:
So the only diff would be that it would lead to non-optimized rust binary?

Reply:
> So the only diff would be that it would lead to non-optimized rust binary?

Correct. That is my understanding after revisiting these options.

Reply:
Are we OK with non-optimized rust binary then? Should I remove `enable_rust_debug` altogether?

Reply:
I'm OK with removing `enable-rust-debug` if the following behaviour is respected:
- A default ./configure will add the --release flag to cargo. This results in an optimized build, but keeps debug symbols.
- `--enable-debug` will remove the --release flag to cargo resulting in a non-optimized debug build.

This does remove some control of Rust vs C build options, but I think its much simpler to reason about.

Thoughts @victorjulien ?

Reply:
@victorjulien how do you feel about this?

Reply:
I'm good with Jasons suggestion.

Reply:
@victorjulien is this already done or not required?

Reply:
I don't understand what your asking. The idea is to implement Jasons idea.